vortex-array/src/array/statistics.rs

```diff
@@ -9,18 +9,31 @@ use crate::{Array, ArrayImpl};
 
 /// Extension functions for arrays that provide statistics.
 pub trait ArrayStatistics {
-    fn is_constant(&self) -> bool;
+    fn is_constant(&self) -> bool {
```

Comment:
Add a docstring that this is best-effort, no canonicalzation. (vs `statistics.compute_is_constant()`)

Reply:
And we should flip the default so the compute function actually does compute